### ISSUE TITLE
hlint pragmas ignored when called via the API

### ISSUE THREAD

**Issue by alanz** [ndmitchell/hlint#351]
As shown in https://github.com/alanz/haskell-ide-engine/tree/hlint-pragmas, when invoking `hlint-2.0.9` with a filename but empty `args` via

```haskell
runLintCmd :: FilePath -> [String] -> EitherT [Diagnostic] IO [Idea]
runLintCmd file args =
  do (flags,classify,hint) <- liftIO $ argsSettings args
     res <- bimapEitherT parseErrorToDiagnostic id $ EitherT $ parseModuleEx flags file Nothing
     pure $ applyHints classify hint [res]
```
on file

```haskell
{-# ANN module ("hlint: ignore Redundant do" :: String) #-}

main = do
  putStrLn "hello"
```

it reports for the redundant do, despite the pragma to the contrary.

**Comment by ndmitchell** [comment]
Note that `ignore` doesn't say don't report, it merely says report at `Ignore` priority. What is the `ideaSeverity` of the returned hint?

**Comment by alanz** [comment]
We are mapping `Ignore` to the LSP `Info` priority. Will filter them out instead. Thanks.

**Comment by alanz** [comment]
FYI, resolved in hie by https://github.com/alanz/haskell-ide-engine/commit/45ffb42f038a34767cb7e58dfb61418575768d0f#diff-7b1a90d13e6fd0644f04b15c6c20befeR145
